### ISSUE TITLE
Detect incorrect matcher use and emit a warning

### ISSUE THREAD

**Issue by TimothyJones** [pact-foundation/pact-js#611]
Sometimes we see people use matchers directly rather than invoking the function:

```
body: {
  id : Matchers.decimal
}
```
We could detect whether functions were passed into the interaction and emit warnings (or errors).

**Comment by mefellows** [comment]
Yep, great idea. I've also created a new label "Developer Experience" to track these sorts of things.
